src/component/Navbar.js

```diff
@@ -1,31 +1,73 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { FaUserCircle } from "react-icons/fa";
 import { Link } from "react-router-dom";
+import { FaBars } from "react-icons/fa";
+
+const Navbar = ({ onLogout }) => {
+  const [userData, setUserData] = useState({});
+  const [showLogoutModal, setShowLogoutModal] = useState(false);
+  const [showMenu, setShowMenu] = useState(false);
+
+  useEffect(() => {
+    const user = JSON.parse(localStorage.getItem("user"));
+    if (user) {
+      setUserData(user);
+    }
+  }, []);
 
-const Navbar = () => {
   return (
-    <div className="flex flex-col md:flex-row justify-between items-center px-6 md:px-20 py-2 bg-blue-400">
-      <ul className="flex-col md:flex hidden md:flex-row gap-5 text-lg font-semibold cursor-pointer md:items-center">
-        <li>
-          <Link to={"/polling"}>Polls</Link>
-        </li>
-        <li>
-          <Link to={"/addpoll"}>Add Poll</Link>
-        </li>
-        <li>
-          <Link to={"/createuser"}>Create User</Link>
-        </li>
-        <li>
-          <Link to={"/users"}>List Users</Link>
-        </li>
-      </ul>
-      <div className="flex items-center gap-2">
-        <div className="text-4xl cursor-pointer">
-          <FaUserCircle />
+    <div className="relative">
+      <div className="flex justify-between items-center px-4 sm:px-8 md:px-10 py-2 bg-black text-white">
+        <div className="md:hidden block text-xl">
+          <FaBars onClick={() => setShowMenu(!showMenu)} />
         </div>
-        <div className="font-semibold">
-          <h1 className="text-base">Amit Ark</h1>
-          <p>amitkumar@innoteach.in</p>
+        <ul
+          className={`absolute bg-black px-4 sm:px-8 pb-4 top-[100%] left-0 ${
+            showMenu ? "flex" : "hidden"
+          } w-full md:pb-0 md:static flex-col md:flex md:flex-row gap-3 text-[14px] md:text-lg font-semibold cursor-pointer md:items-center`}
+          onClick={() => {
+            setShowMenu(false);
+            setShowLogoutModal(false);
+          }}
+        >
+          <li>
+            <Link to={"/polling"}>Polls</Link>
+          </li>
+          {userData.roleId === 2 && (
+            <>
+              <li>
+                <Link to={"/addpoll"}>Add Poll</Link>
+              </li>
+              <li>
+                <Link to={"/createuser"}>Create User</Link>
+              </li>
+              <li>
+                <Link to={"/users"}>List Users</Link>
+              </li>
+            </>
+          )}
+        </ul>
+        <div
+          className="flex items-center gap-2"
+          onClick={() => setShowLogoutModal(!showLogoutModal)}
```

Comment:
Is it for particulat logout modal, I think you are showing name email also so take a state vairable according to it.